### ISSUE TITLE
[Collapsible] Error when using wrapped CollapsibleContent imported from another module

### ISSUE THREAD

**Issue by ryb73** [radix-ui/primitives#2273]
### Current Behavior

A minimal reproduction of the error is set up at https://github.com/ryb73/radix-collapsible-error. You can easily see the error in action by cloning the repo, installing with `yarn`, and running `yarn start`.

First, I have a component that wraps Collapsible.Content:

```ts
import * as Collapsible from "@radix-ui/react-collapsible";

export function CustomContent() {
  return <Collapsible.Content>Content</Collapsible.Content>;
}
```

This component is published (see https://github.com/ryb73/example-collaspible-content) so that it can be used in a React app:

```ts
import * as Collapsible from "@radix-ui/react-collapsible";
import { CustomContent } from "example-collapsible-content";

function App() {
  return (
    <Collapsible.Root>
      <Collapsible.Trigger>Trigger</Collapsible.Trigger>
      <CustomContent />
    </Collapsible.Root>
  );
}

export default App;
```

Starting with `@radix-ui/react-collapsible@1.0.3` I get the following error:
```
Uncaught Error: `CollapsibleContent` must be used within `Collapsible`
    at useContext (createContext.tsx:72:1)
    at Collapsible.tsx:130:1
    at renderWithHooks (react-dom.development.js:16305:1)
    at updateForwardRef (react-dom.development.js:19226:1)
    at beginWork (react-dom.development.js:21636:1)
    at HTMLUnknownElement.callCallback (react-dom.development.js:4164:1)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:4213:1)
    at invokeGuardedCallback (react-dom.development.js:4277:1)
    at beginWork$1 (react-dom.development.js:27451:1)
    at performUnitOfWork (react-dom.development.js:26557:1)
```

### Expected behavior

Prior to version 1.0.3 of `@radix-ui/react-collapsible`, this worked without error. If you cloned https://github.com/ryb73/radix-collapsible-error, you can downgrade to react-collapsible@1.0.2 in order to see a working version of the app.

### Your environment

| Software         | Name(s) | Version |
| ---------------- | ------- | ------- |
| Radix Package(s) | react-collapsible        |  1.0.3       |
| React            | n/a     | 17-18        |
| Browser          | FF, Chrome        |         |
| Assistive tech   |         |         |
| Node             | n/a     |         |
| npm/yarn         | yarn        |  3+       |
| Operating System |  Mac       |  Ventura       |

**Comment by benoitgrelard** [comment]
@ryb73 it sounds like it's because they are both using 2 different versions of the underlying collapsible package, hence getting different context references. Ensure the versions are the same, hence why when you downgrade it works.

**Comment by ryb73** [comment]
Hi @benoitgrelard. That's not what's happening, actually. If you look at the yarn.lock you can see there are no duplicated/mismatched radix-ui packages: https://gist.github.com/ryb73/7b9943ac9e77ff6dd73540dd9c453e4c

**Comment by benoitgrelard** [comment]
hmmm… I'm not sure what it could be then, we rewrap the primitives extensively at WorkOS in our own design system and never have these kinds of context issues.

I will do a bit more digging.

**Comment by benoitgrelard** [comment]
Ok, I think I've figured out the issue, or at least part of it. It seems to be due to some discrepancy between cjs and esm in your setup.

I placed some `console.log` in the node_modules version of the Radix package, in both the cjs and esm exports.

(note: I had to clear a bunch of caches for these edits to be picked up in dev mode, but building it and serving is how I first saw the logs).

What I see is that I get lots from both exports when running your app.

So what I believe is happening is:

- Your app which imports the radix package directly, resolves to the esm export
- You only compiled your "lib" as commonjs, and its package.json also uses just `main`.
- So it looks like your lib resolves the cjs export

Hence why you et 2 different context and the error is raised.

I believe one way to solve this would be to ensure your lib is esm compatible too, or that your app uses cjs all the way.
